internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -62,38 +67,11 @@ func GetTlsConfigurationForServer() *tls.Config {
 	// Create mTLS credentials for server.
 	return &tls.Config {
 		Certificates: []tls.Certificate{cert},
-		VerifyPeerCertificate: verifyPeerCertificateFunc("s2a_test_cert", certPool), // TODO(rmehta19): Call cert verifier library.
+		VerifyPeerCertificate: certverifier.VerifyServerCertificateChain("s2a_test_cert", "host", certPool),
```

Comment:
Similar to the above comment, this should be the VerifyClientCertificateChain API.